hotshot-state-prover/src/service.rs

```diff
@@ -420,6 +422,24 @@ pub async fn sync_state<ApiVer: StaticVersionType>(
         .wallet(wallet)
         .on_http(state.config.provider_endpoint.clone());
 
+    // only sync light client state when gas price is sane
+    if let Some(max_gas_price) = state.config.max_gas_price {
+        let cur_gas_price = provider
+            .get_gas_price()
+            .await
+            .map_err(|e| ProverError::NetworkError(anyhow!("{e}")))?;
+        if cur_gas_price > max_gas_price {
+            tracing::debug!(
+                "Current gas price too high: cur={} gwei, max={} gwei",
+                format_units(cur_gas_price, "gwei")
+                    .map_err(|e| ProverError::Internal(format!("{e}")))?,
+                format_units(max_gas_price, "gwei")
+                    .map_err(|e| ProverError::Internal(format!("{e}")))?,
+            );
```

Comment:
I have a question here, so lets say that LightClient syncs after 12 hours. Currently the has prices are high, will that mean it wont sync for another 12 hours? Should we maybe add a retryInterval instead when there is an error?

Reply:
sure we can retry, but asking it to sleep for 10min is thread blocking? (since this prover service is a single service container). 

I tend to think the cron job ("retry logic") should live in some scripts instead of this prover service logic. 
@Ancient123 @babdor do you have opinions?

Reply:
thanks for the input, updated in [437dbb1](https://github.com/EspressoSystems/espresso-network/pull/3263/commits/437dbb1b2ecc905ede2bf52937fb05a8ce2ba56a)

now we return Err if gas price to high, so daemon mode would retry based on the original (configurable) `retry_interval`, and in the run_once mode, will log a static ERROR msg for easy filtering on datadog.